pkg/rancher-desktop/main/commandServer/settingsValidator.ts

```diff
@@ -53,8 +53,15 @@ export default class SettingsValidator {
   allowedSettings: SettingsValidationMap | null = null;
   allowedTransientSettings: TransientSettingsValidationMap | null = null;
   synonymsTable: settingsLike|null = null;
-
-  validateSettings(currentSettings: Settings, newSettings: RecursivePartial<Settings>): [boolean, string[]] {
+  isKubernetesDesired = false;
+  lockedSettings: LockedSettingsType|null = null;
+
+  validateSettings(
+    currentSettings: Settings,
+    newSettings: RecursivePartial<Settings>,
+    lockedSettings: LockedSettingsType|null): [boolean, string[]] {
+    this.lockedSettings = lockedSettings;
+    this.isKubernetesDesired = typeof newSettings.kubernetes?.enabled !== 'undefined' ? newSettings.kubernetes.enabled : currentSettings.kubernetes.enabled;
```

Comment:
I _think_ this is basically
```suggestion
    this.isKubernetesDesired = newSettings.kubernetes?.enabled ?? currentSettings.kubernetes.enabled;
```
(I don't think `newSettings.kubernetes.enabled` can be `null`)

Reply:
the rewite makes sense